### ISSUE TITLE
Switch project licence to GPL ?

### ISSUE THREAD

**Issue by BenoitPoulet** [NagiosEnterprises/ncpa#642]
Hello,

A lot of Nagios's project are published under the GPL, but not this one.

Can you consider switching to GPL ?

As NRPE is deprecated, it would be nice to have a successor 

Regards.

**Comment by jomann09** [comment]
We have talked about this in the past and I think we will probably be switching in the future, will have to bring it up internally again to see when we can get this done.

**Comment by sni** [comment]
looking at the depenendencies of this project, i am wondering if the current license is compatible with the used librarys in first place?

**Comment by MP-ICT-Diensten** [comment]
Is there any progress on this issue?

Changing the license would make NCPA usable for us, the current license is a blocking issue.

Regards, Marco

**Comment by uSlackr** [comment]
@jomann09  Any updates on this?  It's been almost a year since you last commented

**Comment by jomann09** [comment]
We've talked about it internally a few times, and unless something changes, we should be changing it to a version of GPL for the NCPA 3 release.

**Comment by uSlackr** [comment]
Thanks. Any sense of the timing?  3 months? A year? 😀

\\\\ Greg
________________________________

**Comment by pyther** [comment]
Any update? The current license is very restrictive.

**Comment by ericloyd** [comment]
No one at Nagios is paying any attention to this anymore and I doubt that they'll ever update the license to something truly open source.

**Comment by ChristopherSchultz** [comment]
> No one at Nagios is paying any attention to this anymore and I doubt that they'll ever update the license to something truly open source.

"No true open-sourceman?"

I'm curious about what makes this license to restrictive? Not being able to fork the project? Commercial use of trademarks? Not something your lawyers have reviewed? It's more expensive to build a monitoring system from the ground-up than to pay lawyers to read a 10-paragraph license.

**Comment by ChristopherSchultz** [comment]
Fair enough. They don't bother to define the poison-pills in the license, so I wonder how enforceable that would actually be. But it's clear they are saying "don't use this with Zabbix, etc."

**Comment by ericloyd** [comment]
Even if it allowed for use with other, non-commercial software, or even just standalone use (which it's quite useful for) I'd be happier, personally.

> #6 - The Software may only be used in conjunction with products, projects, and other software distributed by the Company. Any standalone use of the Software, or use of the Software in conjunction with products, projects, or other software not authored or distributed by the Company - including third-party software that competes with the offerings of Company - is strictly prohibited and is a direct violation of the terms of the license.

**Comment by ccztux** [comment]
Beta 3 is out now, yeah! :+1: :tada:
Any updates on this issue?

**Comment by sawolf** [comment]
Hey guys,

I don't have an update for you right now, except to say that I brought it up with @corynorell and with people further up the chain again.

> looking at the depenendencies of this project, i am wondering if the current license is compatible with the used librarys in first place?

@sni, I apologize for missing this the first time; that's interesting to hear. Every dependency I could find had a permissive license. If you can point at a specific dependency where we're not in compliance, that probably changes the internal discussion by quite a bit. However, as far as I can tell, NOSL is technically compatible with using dependencies using public domain/zlib license/apache/BSD/MIT/Python Foundation License.

**Comment by ccztux** [comment]
For everyone who is dissatisfied with the actual license: [SNClient+](https://github.com/ConSol-Monitoring/snclient) could be a good alternative to NCPA.

**Comment by ericloyd** [comment]
The problem is that, like NSClient++ and a few other agents, it's not an official Nagios Enterprises product.  Which means that when the maintainer gets bored of maintaining it, it stops being in sync with any Nagios updates.

Granted, any agent is typically good enough (and if you include SSH, SNMP, WMI, and HTTP requests as "agents" then things get even easier) but NCPA is supposed to be the official Nagios Enterprise agent to take us into the future (go see the 2014 Nagios World Conference clips about its release).  Most of our clients in the government space look at third-party apps as suspect to begin with, so if it doesn't come from Nagios Enterprises, they are hesitant to add it to their mix.

I'll take a look at the software, but if it's NRPE based, then it still suffers from the same problems that early protocol has.

**Comment by BenoitPoulet** [comment]
@ericloyd 
If you want to stay in the Nagios Enterprise scope the current license is not a problem.

The PB is for people would like to use NSPA with something else than Nagios or want to fork it. Which is not possible with this license.

Anyway, the license switch is not important anymore because some alternatives exist, like SNClient+ or Monitoring agent. And the paradigm of monitoring is moving toward other methods.

The only impact was for Nagios enterprise, who is losing more and more appeal : nobody is talking anymore of Nagios is the observability world.

**Comment by ericloyd** [comment]
You're right.  I got side tracked there for a minute.  I don't see the solution to the NCPA license issue being "switch to a different product."  My complaint is that open source software should be open to be used freely, not restricted to only being in conjunction with the author's other products.

So for me, I'm not looking for alternatives.  I'm looking for a change to the existing license.

Be that as it may, 100% of our company's revenue comes from either selling Nagios licenses or providing Nagios consulting services and I'm not sure I agree with your statement that "nobody is talking anymore of Nagios" but that's a topic for another day.  :-)

**Comment by nook24** [comment]
> My complaint is that open source software should be open to be used freely, not restricted to only being in conjunction with the author's other products.

Saems like Nagios is shifting away from open source. Like in this comment: https://github.com/NagiosEnterprises/ndoutils/issues/57#issuecomment-596603014

We have build our own monitoring solution and agent, all open (and Nagios Core compatible)
- https://github.com/it-novum/openITCOCKPIT
- https://github.com/it-novum/openitcockpit-agent-go

So you can still get everything from a single vendor and of course, use any other agent you like to use

**Comment by ericloyd** [comment]
Not arguing about walled gardens.  Simply talking about NCPA being published as open source and then having a restrictive license.  Those two things are not compatible. Pick one, Nagios! :-)
